next.config.js

```diff
@@ -1,49 +1,27 @@
-const path = require('path')
+const withPreact = require('@zeit/next-preact')
 const SWPrecacheWebpackPlugin = require('sw-precache-webpack-plugin')
-const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin
 
-module.exports = {
-  webpack: function (config, { dev }) {
-    if (dev) {
-      return config
-    }
-
-    config.plugins.push(
-      new BundleAnalyzerPlugin({
-        analyzerMode: 'disabled',
-        // For all options see https://github.com/th0r/webpack-bundle-analyzer#as-plugin
-        generateStatsFile: true,
-        // Will be available at `.next/stats.json`
-        statsFilename: 'stats.json'
-      })
-    )
-
-    const oldEntry = config.entry
-
-    config.entry = () => oldEntry()
-      .then(entry => {
-        entry['main.js'].push(
-          path.resolve('./utils/offline.js'),
-          path.resolve('./utils/track.js')
-        )
+const returnConfig = (config, options, nextConfig = {}) => {
+  if (typeof nextConfig.webpack === 'function') {
+    return nextConfig.webpack(config, options)
+  }
 
-        entry.commons = ['./utils/prismTemplateString.js']
-        return entry
-      })
+  return config
+}
 
-    config.resolve.alias = config.resolve.alias || {}
+const config = (nextConfig = {}) => Object.assign({}, nextConfig, {
+  webpack(config, options) {
+    const { dev, isServer } = options
+    if (dev || !isServer) {
+      return returnConfig(config, options, nextConfig)
+    }
 
     config.plugins.push(
       new SWPrecacheWebpackPlugin({
```

Comment:
Maybe want to check https://github.com/hanford/next-offline out to see whether it offers anything on top or simplifies the conf?

Reply:
Doesn’t seem like it actually improves anything :/

It’s modifications to the webpack config are pretty similar to the ones I had to remove due to some changes that the Next team made

Reply:
Configuring by hand at least is transparent, keep it as it is.